provider/image.go

```diff
@@ -419,10 +419,18 @@ func marshalCachedImages(b resource.PropertyValue) ([]string, error) {
 	// if we specify a list of stages, then we only pull those
 	cacheFrom := c.ObjectValue()
 	if cacheFrom["images"].IsNull() {
-		return cacheImages, fmt.Errorf("if you want to use cacheFrom, you must have `images` set")
+		return cacheImages, fmt.Errorf("cacheFrom requires an `images` field")
```

Comment:
What happens if `cacheFrom` is known but `cacheFrom["images"]` is unknown in preview?

Essentially, I can't find this test case:
```go
	t.Run("Test Cached Images Passes On Unknowns", func(t *testing.T) {
		expected := []string(nil)
		buildInput := resource.NewObjectProperty(resource.PropertyMap{
			"cacheFrom": resource.NewObjectProperty(resource.PropertyMap{
				"images": resource.NewNullProperty(), // unknowns are passed as null property values
			}),
		})
		actual, err := marshalCachedImages(buildInput)
		assert.NoError(t, err)
		assert.Equal(t, expected, actual)
	})
```

Reply:
Nice catch! I changed it to traditional key presence checking, and return nil on an unknown field.

Reply:
Nice catch! I changed it to traditional key presence checking, and return nil on an unknown field.

Reply:
Nice catch! I changed it to traditional key presence checking, and return nil on an unknown field.